Objects/genericaliasobject.c

```diff
@@ -626,6 +640,65 @@ static PyNumberMethods ga_as_number = {
         .nb_or = _Py_union_type_or, // Add __or__ function
 };
 
+static PyObject *
+ga_iternext(gaiterobject *gi) {
+    gaobject *alias = (gaobject *) gi->obj;
+
+    if (gi->exhausted) {
+        gi->obj = NULL;
+        return NULL;
+    }
+    gi->exhausted = 1;
+
+    gaobject *starred_tuple = PyObject_GC_New(gaobject, &Py_GenericAliasType);
+    starred_tuple->origin = alias->origin;
+    starred_tuple->args = alias->args;
+    starred_tuple->parameters = alias->parameters;
+    starred_tuple->weakreflist = alias->weakreflist;
```

Comment:
I wonder if ``*starred_tuple = *alias`` works here instead of copying the fields individually. Probably not.